opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -125,12 +150,9 @@ impl<T: Number<T>> Sum<T> {
         s_data.is_monotonic = self.monotonic;
         s_data.data_points.clear();
 
-        let mut values = match self.value_map.values.write() {
-            Ok(v) => v,
-            Err(_) => return (0, None),
-        };
-
-        let n = values.len() + 1;
+        // Max number of data points need to account for the special casing
+        // of the no attribute value + overflow attribute.
+        let n = self.value_map.count.load(Ordering::Relaxed) + 2;
```

Comment:
We should update this calculation for cumulative as well?

Reply:
yes, but it is not a mandatory step, as the vec will just grow if pre-allocated size is not big enough. In cumulative, once grown, it remains in that size forever, so not much of an issue.
Will make the change to make it consistent, but this is an area (exporting), that needs revisit/refactor.